sdk/search/Azure.AI.Search/src/SearchExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+
+namespace Azure.AI.Search
+{
+    /// <summary>
+    /// Search extension methods.
+    /// </summary>
+    internal static partial class SearchExtensions
+    {
+        /// <summary>
+        /// Assert that the given URI uses HTTPS as its scheme.
+        /// </summary>
+        /// <param name="endpoint">The URI to validate.</param>
+        /// <param name="paramName">
+        /// The name of the parameter for this URI, to use with an
+        /// <see cref="ArgumentException"/>.
+        /// </param>
+        /// <exception cref="ArgumentException">
+        /// Thrown when the <paramref name="endpoint"/> is not using HTTPS as
+        /// its scheme.
+        /// </exception>
+        public static void AssertHttpsScheme(this Uri endpoint, string paramName)
```

Comment:
Tip: since this is asserting arguments (as used), you can actually "extend" the `Arguments` validator class for consistency and discoverability: https://azure.github.io/azure-sdk/dotnet_implementation.html#dotnet-parameters

Reply:
I actually want to move this (and a few other things) up into the shared source in a future PR.  I think we'll all be doing HTTPS validation so it would make sense to throw the same exception here, throw the same exception for version validation, etc.

Reply:
In that case, probably just add it to the primary Argument.cs file.